addons/mail/static/src/core/common/chat_hub_model.js

```diff
@@ -51,10 +51,8 @@ export class ChatHub extends Record {
     actuallyHidden = Record.many("ChatWindow");
 
     closeAll() {
-        while (this.opened.length > 0 || this.folded.length > 0) {
-            this.opened[0]?.delete();
-            this.folded[0]?.delete();
-        }
+        const cws = [...this.opened, ...this.folded];
+        cws.forEach((cw) => cw.close());
```

Comment:
This could also be a single line, which would prevent the spread of these hard to read variables.

Reply:
`[...this.opened, ...this.folded].forEach((cw) => cw.close());`